src/StratumClient/StratumClient.fs

```diff
@@ -29,6 +30,7 @@ type StratumClient(endpoint: string, port: int) =
     inherit Client(endpoint, port)
 
     static member Deserialize<'T> json: 'T =
+        if String.IsNullOrWhiteSpace(json) then failwith "Invalid json format, the server response was empty"
```

Comment:
shouldn't you throw an ex that inherits from ConnectionUnsuccessfulException? otherwise this would break the tests

Reply:
I saw this as an edge case so it should break hard, but sure that makes sense. Maybe anything thrown from the deserialization should be wrapped in a ConnectionUnsuccessfulException (or derivative)?

Reply:
actually you're right, if we break hard for malformed json, we should break hard here too

Reply:
So you want to wrap all deserialization errors or let it be?

Reply:
let it be